core/templates/pages/skill-editor-page/editor-tab/skill-concept-card-editor/skill-concept-card-editor.directive.spec.ts

```diff
@@ -0,0 +1,188 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for the skill concept card editor directive.
+ */
+
+// TODO(#7222): Remove the following block of unnnecessary imports once
+// the code corresponding to the spec is upgraded to Angular 8.
+import { UpgradedServices } from 'services/UpgradedServices';
+// ^^^ This block is to be removed.
+
+import { EventEmitter } from '@angular/core';
+
+describe('Skill concept card editor directive', function() {
+  var $scope = null;
+  var ctrl = null;
+  var $rootScope = null;
+  var $uibModal = null;
+  var directive = null;
+  var QuestionCreationService = null;
+  var SkillUpdateService = null;
+  var $q = null;
+  var SkillEditorStateService = null;
+  var WindowDimensionsService = null;
+  var SkillObjectFactory = null;
+  var misconceptionDict = {
+    feedback: 'feedback',
+    id: 'id1',
+    must_be_addressed: false,
+    name: 'name1',
+    notes: 'notes1'
+  };
+  var rubricDict = {
+    difficulty: 'Easy',
+    explanations: ['Easy']
+  };
+
+  var conceptCardDict = {
+    explanation: {content_id: 'content',
+      html: 'html_data'},
+    worked_examples: [],
+    recorded_voiceovers: {
+      voiceovers_mapping: {
+        explanation: {},
+        worked_example_1: {},
+        worked_example_2: {}
+      }
+    } };
+  var skillBackendDict = {
+    all_questions_merged: true,
+    description: 'description1',
+    id: 'skillId1',
+    language_code: 'en',
+    misconceptions: [misconceptionDict],
+    next_misconception_id: '2',
+    prerequisite_skill_ids: [],
+    rubrics: [rubricDict],
+    skill_contents: conceptCardDict,
+    superseding_skill_id: 'skillId2',
+    version: 2,
+  };
+  beforeEach(angular.mock.module('oppia'));
+  beforeEach(angular.mock.module('oppia', function($provide) {
+    var ugs = new UpgradedServices();
```

Comment:
You shouldn't need this right? Use testbed.get instead

Reply:
Hey @srijanreddy98 Is this fine or should i remove this and use TestBed?

Reply:
Follow this: https://github.com/oppia/oppia/pull/10659